pkgs/development/libraries/libdatachannel/default.nix

```
@@ -0,0 +1,61 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, ninja
+, pkg-config
+, libnice
+, nlohmann_json
+, openssl
+, srtp
+, usrsctp
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libdatachannel";
+  version = "0.18.4";
+
+  src = fetchFromGitHub {
+    owner = "paullouisageneau";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-kSK+5gFMG6tq89R1m08gNBKPdwyR/mLEDhWXQ/uk34o=";
+    fetchSubmodules = true;
+  };
+
+  outputs = [ "out" "dev" ];
+
+  strictDeps = true;
+  nativeBuildInputs = [
+    cmake
+    ninja
+    pkg-config
+  ];
+  buildInputs = [
+    libnice
+    nlohmann_json
+    openssl
+    srtp
+    usrsctp
+  ];
+
+  cmakeFlags = [
+    "-DUSE_NICE=ON"
+    "-DUSE_SYSTEM_JSON=ON"
+    "-DUSE_SYSTEM_SRTP=ON"
+    "-DUSE_SYSTEM_USRSCTP=ON"
```

```suggestion
    "-DUSE_NICE=ON"
    "-DUSE_SYSTEM_SRTP=ON"
    "-DNO_EXAMPLES=ON"
```
- USE_SYSTEM_JSON/USE_SYSTEM_USRSCTP only exist on the master branch for now
- I'd skip building the examples for now (since it currently requires a vendored nlohmann_json and we don't install them anyway)